File: components/image_fetcher/core/request_metadata.cc
// Copyright 2017 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "components/image_fetcher/core/request_metadata.h"

namespace image_fetcher {

RequestMetadata::RequestMetadata()
    : http_response_code(RESPONSE_CODE_INVALID) {}

bool operator==(const RequestMetadata& lhs, const RequestMetadata& rhs) {
  return lhs.mime_type == rhs.mime_type &&
         lhs.http_response_code == rhs.http_response_code &&
         lhs.content_location_header == rhs.content_location_header;
}

bool operator!=(const RequestMetadata& lhs, const RequestMetadata& rhs) {
  return !(lhs == rhs);
}

}  // namespace image_fetcher

File: components/image_fetcher/android/java/src/org/chromium/components/image_fetcher/RequestMetadata.java
// Copyright 2025 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.components.image_fetcher;

import org.jni_zero.JNINamespace;

import org.chromium.build.annotations.NullMarked;

@JNINamespace("image_fetcher")
@NullMarked
public class RequestMetadata {
    public final String mimeType;
    public final int httpResponseCode;
    public final String contentLocationHeader; // Corresponds to the fields of C++ RequestMetadata.

    public RequestMetadata(String mimeType, int httpResponseCode, String contentLocationHeader) {
        this.mimeType = mimeType;
        this.httpResponseCode = httpResponseCode;
        this.contentLocationHeader = contentLocationHeader;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("Image fetcher request metadata: httpResponseCode = ");
        sb.append(httpResponseCode);
        sb.append(", mimeType = ");
        sb.append(mimeType);
        sb.append(", contentLocationHeader = ");
        sb.append(contentLocationHeader);
        return sb.toString();
    }
}
